packages/core/src/codewhisperer/util/supplementalContext/utgUtils.ts

```diff
@@ -24,10 +24,12 @@ import { getOpenFilesInWindow } from '../../../shared/utilities/editorUtilities'
 import { getLogger } from '../../../shared/logger/logger'
 import { CodeWhispererSupplementalContext, CodeWhispererSupplementalContextItem, UtgStrategy } from '../../models/model'
 
-type UtgSupportedLanguage = keyof typeof utgLanguageConfigs
+const UTG_SUPPORTEDLANGUAGES: vscode.TextDocument['languageId'][] = ['java', 'python']
```

Comment:
Can we fetch the list of supported languages to BE. Because In order to extend the feature to additional languages we will rely on weekly release of IDEs, instead we should rely on Backend for this information so that we can make such decisions very dynamic.

Reply:
Yea that's a good callout and discussed offline with Vikash, I think we're limited by the infra we have at this point of time, our service doesn't have this kinda of dynamic configuration API yet except the one for A/B testing. This comment also apply to other Q features as well like "auto trigger classifier co-efficients" https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/codewhisperer/service/classifierTrigger.ts#L43  and other language level gated features.